webapp/src/client/client.js

```diff
@@ -263,62 +265,79 @@ export default class Client {
-    doGet = async (url, headers = {}, credentials) => {
-        headers.Accept = 'application/json';
-        let options = {
-            method: 'get',
-            headers,
-        };
+    
+    doGet = (url, headers = {}, credentials) => {
 
-        if (url.includes('plugins/' + pluginID)) {
-            options = Client4.getOptions(options);
-        }
+        return async (dispatch, getState) => {
+            const baseUrl = getPluginServerRoute(getState()) + "/"+ url;
+            headers.Accept = 'application/json';
+            let options = {
+                method: 'get',
+                headers,
+            };
 
-        if (credentials) {
-            options.credentials = credentials;
-        }
+            if (url.includes('plugins/' + pluginID)) {
+                options = Client4.getOptions(options);
+            }
 
-        const response = await fetch(url, options);
+            if (credentials) {
+                options.credentials = credentials;
+            }
 
-        if (response.ok) {
-            return response.json();
-        }
+            const response = await fetch(baseUrl, options);
 
-        const text = await response.text();
+            if (response.ok) {
+                return response.json();
+            }
 
-        throw new ClientError(Client4.url, {
-            message: text || '',
-            status_code: response.status,
-            url,
-        });
-    }
+            const text = await response.text();
+
+            throw new ClientError(Client4.url, {
+                message: text || '',
+                status_code: response.status,
+                url,
+            });
 
-    doPost = async (url, body, headers = {}) => {
-        headers.Accept = 'application/json';
-        headers['Content-Type'] = 'application/json';
-        let options = {
-            method: 'post',
-            body: JSON.stringify(body),
-            headers,
+            //return true;
         };
 
-        if (url.includes('plugins/' + pluginID)) {
-            options = Client4.getOptions(options);
-        }
 
-        const response = await fetch(url, options);
 
-        if (response.ok) {
-            return response.json();
-        }
+      
+    }
 
-        const text = await response.text();
+    doPost =  (url, body, headers = {}) => {
+         return async (dispatch, getState) => {
+             const baseUrl = getPluginServerRoute(getState()) + "/" + url;
+             headers.Accept = 'application/json';
+             headers['Content-Type'] = 'application/json';
+             let options = {
+                 method: 'post',
+                 body: JSON.stringify(body),
+                 headers,
+             };
 
-        throw new ClientError(Client4.url, {
-            message: text || '',
-            status_code: response.status,
-            url,
-        });
+             if (url.includes('plugins/' + pluginID)) {
+                 options = Client4.getOptions(options);
```

Comment:
Same suggestion here regarding applying the server URL to Mattermost-specific URLs only:

```suggestion
                 url = getPluginServerRoute(getState()) + "/"+ url;
                 options = Client4.getOptions(options);
```

Then we can remove line 311